### PR TITLE
[Example/OpenCV] Add OpenCV custom filter example: Average

### PR DESCRIPTION
**Pull request by again4you**
This patch newly adds an OpenCV custom filter example, which calculates
the average value of input tensor for each channel. The shape of the
input tensor is [N][y][x][M] and that of the output tensor is [N][1][1][M].

This example is developed based on `nnstreamer_customfilter_example_average.c` file.

* Resolves: #678 

**Self evaluation:**
1. Build test: [*]Passed [ ]Failed []Skipped
2. Run test: [*]Passed [ ]Failed []Skipped

Signed-off-by: Sangjung Woo <sangjung.woo@samsung.com>

### CODE REVIEW

File: nnstreamer_example/custom_example_opencv/nnstreamer_customfilter_opencv_average.cc [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+ * @brief set_inputDim
+ */
+static int
+set_inputDim (void *private_data, const GstTensorFilterProperties * prop,
+    const GstTensorsInfo * in_info, GstTensorsInfo * out_info)
+{
+  int i;
+  pt_data *pdata = static_cast<pt_data *> (private_data);
+
+  assert (pdata);
```
Comment (jaeyun-jung) [inline nnstreamer_example/custom_example_opencv/nnstreamer_customfilter_opencv_average.cc]: If including glib in example app, please consider to use g_assert instead assert.

File: nnstreamer_example/custom_example_opencv/nnstreamer_customfilter_opencv_average.cc [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+ * @author  Sangjung Woo <sangjung.woo@samsung.com>
+ * @bug  No known bugs
+ * @see  nnstreamer_customfilter_example_average.c
+ * 
+ * This example calculates the average value of input tensor for 
+ * each channel (i.e. R, G & B). The shape of the input tensor is
+ * [N][y][x][M] and that of the output tensor is [N][1][1][M]. 
+ */
+
+#include "opencv2/opencv.hpp"
```
Comment (wooksong) [inline nnstreamer_example/custom_example_opencv/nnstreamer_customfilter_opencv_average.cc]:
```suggestion
#include <opencv2/opencv.hpp>
```

However, this is very trivial thing.

File: tests/nnstreamer_filter_custom/runTest.sh [hunk 3]
```diff
@@ -91,4 +91,15 @@ gstTest "--gst-plugin-path=${PATH_TO_PLUGIN} videotestsrc num-buffers=1 ! video/
...
+# Test average using OpenCV (15)
+# cutom version
+PATH_TO_MODEL_A="../../build/nnstreamer_example/custom_example_average/libnnstreamer_customfilter_average.so"
+gstTest "--gst-plugin-path=${PATH_TO_PLUGIN} videotestsrc num-buffers=1 ! video/x-raw,format=RGB,width=640,height=480,framerate=0/1 ! videoconvert ! video/x-raw, format=RGB ! tensor_converter ! tensor_filter framework=\"custom\" model=\"${PATH_TO_MODEL_A}\" ! filesink location=\"testcase15.average.log\" sync=true" 15
+
+# OpenCV version
+PATH_TO_MODEL_A="../../build/nnstreamer_example/custom_example_opencv/libnnstreamer_customfilter_opencv_average.so"
+gstTest "--gst-plugin-path=${PATH_TO_PLUGIN} videotestsrc num-buffers=1 ! video/x-raw,format=RGB,width=640,height=480,framerate=0/1 ! videoconvert ! video/x-raw, format=RGB ! tensor_converter ! tensor_filter framework=\"custom\" model=\"${PATH_TO_MODEL_A}\" ! filesink location=\"testcase15.opencv.average.log\" sync=true" 15
+
+compareAll testcase15.opencv.average.log testcase15.average.log 15
```
Comment (myungjoo) [inline tests/nnstreamer_filter_custom/runTest.sh]: The case number (the last argument) would be better if it's 16 or 15-2.

Comment (myungjoo) [reply tests/nnstreamer_filter_custom/runTest.sh]: Never mind. It's ok with 15.

However, if #729 gets merged before this, you will need to rebase and change a bit anyway.

File: nnstreamer_example/custom_example_opencv/nnstreamer_customfilter_opencv_average.cc (hunk 2 above)
Comment (myungjoo) [reply nnstreamer_example/custom_example_opencv/nnstreamer_customfilter_opencv_average.cc]: @again4you PTAL.

File: nnstreamer_example/custom_example_opencv/nnstreamer_customfilter_opencv_average.cc [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+ */
+typedef struct _pt_data
+{
+  uint32_t in_height;  /***< height of input tensor */
+  uint32_t in_width;   /***< width of input tensor */
+  uint32_t in_channel; /***< channel of input tensor */
+} pt_data;
+
+/**
+ * @brief pt_init
```
Comment (leemgs) [inline nnstreamer_example/custom_example_opencv/nnstreamer_customfilter_opencv_average.cc]: BTW, Could you describe @brief tag in more detail? the annotation is too short. :(
```bash
@brief   summarize a goal/meaning of the function

Describte in detail ASAP.............................................

.....................................................................

.....................................................................
```

File: tests/nnstreamer_filter_custom/runTest.sh [hunk 5]
```diff
@@ -91,4 +91,15 @@ gstTest "--gst-plugin-path=${PATH_TO_PLUGIN} videotestsrc num-buffers=1 ! video/
 python checkScaledTensor.py testcase14.direct.log 640 480 testcase14.scaled.log 1920 1080 3
 casereport 14 $? "Golden test comparison"
 
+# Test average using OpenCV (15)
+# cutom version
```
Comment (leemgs) [inline tests/nnstreamer_filter_custom/runTest.sh]: s/cutom/custom/

File: nnstreamer_example/custom_example_opencv/nnstreamer_customfilter_opencv_average.cc (hunk 4 above)
Comment (myungjoo) [reply nnstreamer_example/custom_example_opencv/nnstreamer_customfilter_opencv_average.cc]: All nnstreamer-custom-filter has the same function lists and this is one of them. We'd better provide some "custom filter code generator" and add good doxygen statements in that code generator. So, it's ok not to have good doxygen brief entries for this common callback.

**Pull request merged**